linode/manager
# feat: [M3-6159] - Improve Linode disk downsize error messaging

## Description 📝

**What does this PR do?**
Increases the amount of feedback provided to users on Linode disk resize actions, particularly failed disk resizes.

Summary of changes:
- Adds helper text to the `LinodeDiskDrawer` to provide (1) a link to documentation on disk resizing and (2) a text tooltip for command to find minimum disk size. The goal of these changes is to provide more context for the user on disk resizing limits to help mitigate a resize error in the first place.
- Adds `HelpIcon` tooltip for disabled Add a Disk button to replace tooltip on disabled button hover to make presence of tooltip more visible to the user. 
- Adds `HelpIcon` tooltip for disabled Resize button to make the reason for disabling clearer: the Linode must be powered off. There is already a tooltip on the Delete action, but the Resize action was missing this context, requiring a user to guess why Resize was unavailable for a powered on Linode.
- Adds persistent error `Toast` notifying user of failed disk resize. The goal of this is to provide more visible error messaging to the user when a disk cannot be resized. (Note: This was changed from an error `Notice` in the mocks to align with our current method of displaying failed event errors.) Previously, the only visible feedback was in the notification menu with an event message on failure of the`disk_resize` action; no error message was returned on the screen itself.
- Update `disks_resize` failure event message in notification menu/events landing to include a docs link about disk resizing. This same doc is also linked in the error `Toast`. 
- Updates tests to after moving some event message logic around and adding the ability to append text to an event message when generated. 
- Creates new `Code` component (used in "Check disk usage" text tooltip for determining minimum disk size) to apply to some styling to text that is a Linux command.

TODO: (We can start reviewing without these completed.)
- [x] Finalize docs links
- [ ] Finalize copy 
- [x] Add GA tags

## Preview 📷
Tooltips:
![Screenshot 2023-03-13 at 10 41 30 AM](https://user-images.githubusercontent.com/114685994/224785165-46724d3a-9c13-436c-b53b-ffc2b9306dc8.jpg)

Disk drawer copy updates:
![Screenshot 2023-03-13 at 10 37 23 AM](https://user-images.githubusercontent.com/114685994/224783409-3cc1bf85-69b7-4c40-b064-e4926ea63def.jpg)

Error messages:
![Screenshot 2023-03-13 at 10 32 34 AM](https://user-images.githubusercontent.com/114685994/224782307-0e159973-0029-4109-81b9-62fdd9a83f45.jpg)


## How to test 🧪

**What are the steps to reproduce the issue or verify the changes?**
1. Have at least one Linode on your account. For the purposes of verifying step 3, using a nanode is probably easiest.
2. Navigate to the Storage tab of the Linode Details page of a powered on Linode. 
3. Verify that the help icon next to the disabled Add a Disk button displays a tooltip about disk space. Verify that hovering over the button no longer displays this tooltip.
4. Verify that the help icon next to the disabled Resize button displays a tooltip about powering down the Linode. This copy should match the tooltip copy for the Delete action.
5. Power down the Linode (or navigate to the details page for a different Linode that is already off).
6. Verify that the Resize action is no longer disabled and does not display a help icon next to it. 
7. Click Resize and verify the copy changes and text tooltip in the Resize drawer match the internal mocks and the docs link redirects as expected. 
8. Change the size to something that will be below the minimum size possible for the disk (e.g. `1 MB`) and submit the resize. 
9. Wait for the resize action to attempt to complete. Verify that when the resize fails, a *persistent* error toast displays. Verify the link in the error toast redirects to a docs page on resizing a disk.
10. Check the notification menu, the events landing page, and the Activity Feed tab (on the linode details page) to verify that an event displays for the resize failure and that it contains a valid docs link that also redirects to the docs page on resizing a disk.

**How do I run relevant unit or e2e tests?**
```
yarn test Event 
```
